pkg/clusteragent/admission/patch/patch_request.go

```diff
@@ -63,8 +72,8 @@ func (k K8sTarget) String() string {
 }
 
 func (k K8sTarget) validate(clusterName string) error {
-	if k.ClusterName != clusterName {
-		return fmt.Errorf("target cluster name %q is different from the local one %q", k.ClusterName, clusterName)
+	if k.Cluster != clusterName {
+		return fmt.Errorf("target cluster name %q is different from the local one %q", k.Cluster, clusterName)
```

Comment:
We're considering this an error because the remote config predicate logic (once implemented) should ensure that this cluster agent only receives updates for this cluster name, right?

Reply:
Correct, if this happens it means there is a bug in RC.